### ISSUE TITLE
Stacking order bug causes color-contrast to misidentify background

### ISSUE THREAD

**Issue by WilcoFiers** [dequelabs/axe-core#4540]
The following should pass:

```html
<style>
  #container {
    position: fixed;
    background-color: #333;
    padding: 0.5em 2em;
  }
  #target {
    filter: contrast(100%);
    color: #eee;
  }
</style>
<div id="container">
  <span id="target"> Hello world </span>
</div>
```

There's some bug in `createGrid()` which results in the stacking order of `#target` to be calculated incorrectly. This comes from the `filter` property. Removing that avoids the issue. This is going to require a very close read of the CSS spec to figure out.

**Comment by padmavemulapati** [comment]
Verified with the latest axe-core develop code base, `color-contrast` violation is not going into false negative and correctly determines that one of the ancestor container elements is rendering on top of the leaf descendant element

not in `incomplete` but in `violations` with proper stack order

https://github.com/user-attachments/assets/d5c31f0c-1a23-4107-bcd6-5552d226d38a

as well, with the code snippet:

```

<div id="1" style="position: relative; z-index: 4; width: 200px; height: 200px;">
          <div id="2" style="position: absolute;">
            <div id="3" style="transform: translate3d(0px, 0px, 0px);">
              <span id="target">text</span>
            </div>
          </div>
        </div>
  
        <div style="position: relative; z-index: 1; width: 200px; height: 200px;">
  <div style="position: absolute;">
    <div style="transform: translate3d(0px, 0px, 0px);">
      <span id="fail13" style="background-color: #111; color: #000">low contrast</span>
      <span id="pass24" style="background-color: #fff; color: #000">ok contrast</span>
    </div>
  </div>
</div>

<div style="position: relative; z-index: 1; width: 200px; height: 200px;">
  <div style="position: absolute;">
    <div style="opacity: 0.99;">
      <span id="fail14" style="background-color: #111; color: #000">low contrast</span>
      <span id="pass25" style="background-color: #fff; color: #000">ok contrast</span>
    </div>
  </div>
</div>

<body style="background-color: #121212; color: #333">
  <div style="position: fixed; width: 100%; height: 100vh;">
    <svg style="position: absolute">
      <g>
        <g transform="translate(212.26931762695312, 0) scale(1)">
          <path stroke="#9e72c9" fill="#7435b2" d=" m 0,0  h 15  l 6,4  3,0  6,-4  h 133.06036376953125  v 5  V 5  V 29  V 29  V 34  h -133.06036376953125  l -6,4  -3,0  -6,-4  h -15  H 0 z
            "></path>
          <g transform="translate(10,9.5)">
            <text fill="#fff" dominant-baseline="central" x="0" y="7.5">play</text>
          </g>
        </g>
      </g>
    </svg>
  </div>
</body>
<style>
  #container {
    position: fixed;
    background-color: #333;
    padding: 0.5em 2em;
  }
  #target {
    filter: contrast(100%);
    color: #eee;
  }
</style>
<div id="container">
  <span id="target"> Hello world </span>
</div>
 
</html>
```


<img width="1329" height="1002" alt="Image" src="https://github.com/user-attachments/assets/438707e3-e1bb-4be1-987a-acb77f3cc6c1" />
